client/cypress/integration/add-user.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { User } from 'src/app/users/user';
+import { AddUserPage } from '../support/add-user.po';
+
+describe('Add user', () => {
+  const page = new AddUserPage();
+
+  beforeEach(() => {
+    page.navigateTo();
+  });
+
+  after(() => {
+    // Lets put the db back to its original state before we leave.
+    //cy.task('seed:database');
+  });
+
+  it('Should have the correct title', () => {
+    page.getTitle().should('have.text', 'New User');
+  });
+
+  it('Should enable and disable the add user button', () => {
+    // ADD USER button should be disabled until...
+    page.addUserButton().should('be.disabled');
+    page.typeInput(cy.get('#nameField'), 'test');
+    page.addUserButton().should('be.disabled');
+    page.typeInput(cy.get('#ageField'), '20');
+    page.addUserButton().should('be.disabled');
+    page.typeInput(cy.get('#emailField'), 'invalid');
+    page.addUserButton().should('be.disabled');
+    page.typeInput(cy.get('#emailField'), 'user@example.com');
+    // all the required fields have valid input, then it should be enabled
+    page.addUserButton().should('be.enabled');
+  });
+
+  describe('Adding a new user', () => {
+
+    const user: User = {
+      _id: null,
+      name: 'Test User',
+      age: 30,
+      company: 'Test Company',
+      email: 'test@example.com',
+      role: 'editor'
+    };
+
+    beforeEach(() => {
+      cy.task('seed:database');
+    });
+
+    it('Should go to the right page, and have the right info', () => {
+      page.addUser(user);
+
+      cy.url()
+        .should('match', /.*\/users\/[0-9a-fA-F]{24}$/)
+        .should('not.match', /.*\/users\/new$/);
```

Comment:
```suggestion
        .should('match', /\/users\/[0-9a-fA-F]{24}$/)
        .should('not.match', /\/users\/new$/);
```

Reply:
I feel like we could be more specific here and actually try to enforce that the _path_ should be exactly `/users\/[0-9a-fA-F]{24}$` but without doing that this is better than before.